enhancements/baremetal/an-slo-for-baremetal.md

```diff
@@ -0,0 +1,592 @@
+
+* A Provisioning resource manifest
+* Manifests for BareMetalHost resources and their associated secrets
+
+and these manifests are applied by the cluster-bootstrap component
+towards the end of the cluster bootstrapping process.
+
+This resource creation step does not succeed until the step (2)
+completes - i.e. the relevant CRDs are applied - and this is currently
+done by the Cluster Version Operator (CVO) as it applies the manifests
+for the MAO.
+
+Finally, (3) happens when the MAO detects that it is running on the
+`baremetal` infrastructure platform and instantiates a relatively
+complex Deployment including the BareMetalHost controller and various
+containers running other components, all from the Metal3 project. The
+configuration of this deployment is driven by the Provisioning
+resource created by the installer in (1).
+
+There are two problems emerging with this design:
+
+* A sense that the MAO is responsible for a significant amount of bare
+  metal specific tasks that should be declared outside of the scope of
+  the MAO, particularly since it does not have equivalent
+  responsibilities on any other platform.
+
+* Expanding needs for bare metal specific manifests - for example, new
+  CRDs used to drive new baremetal capabilities - to be installed
+  early in the cluster bring-up means introducing yet more bare metal
+  specific concerns into the MAO.
+
+Steps (2) and (3) are aspects of cluster bring-up which the CVO is
+clearly well-suited. However, to date, it was understood that creating
+a Second Level Operator (SLO) (in other words, a CVO-managed operator)
+for bare metal would not make sense, since it implied a component
+installed and running on clusters where it is not needed.
+
+### Goals
+
+Allow bare metal machine management capabilities to be fully
+encapsulated in a new SLO.
+
+Ensure that this new SLO has minimal impact on clusters not running on
+the `baremetal` platform.
+
+### Non-Goals
+
+### Proposal
+
+Recognizing that bare metal support warrants the creation of a new
+"subsystem":
+
+1. Create a new, OpenShift-specific project called
+   `cluster-baremetal-operator` aka CBO.
+2. This project should implement a new SLO which is responsible for
+   installing bare metal specific CRDs and running the BareMetalHost
+   controller and related Metal3 components.
+3. The CBO should meet all of the standard expectations of an SLO,
+   including for example keeping a ClusterOperator resource updated.
+4. On infrastructure platforms other than `baremetal`, the CBO should
+   following emerging patterns for "not in use" components, for
+   example setting its cluster operator condition to `Disabled=true`,
+   `Available=true`, and `Progressing=False`.
+
+### Risks and Mitigations
+
+(FIXME)
+
+- Impact on non-baremetal platforms
+- Implementation cost and complexity
+- Upgrades
+- API stability
+- Additional bare metal specific functionality in future
+- Upstream (Metal3) implications
```

Comment:
This section probably needs either some more details or cleanup - I think the most critical items are covered below e.g "Not In Use" SLO Behaviors" for impact on non-baremetal platforms and the discussion around upgrade strategy.